src/core/messenger/background.ts

```diff
@@ -244,6 +244,24 @@ const setupBackground = async (
 					return undefined;
 				};
 
+				let played = false;
+				video.onended = () => (played = true);
+
+				const observer = new IntersectionObserver(
+					(entries) => {
+						entries.forEach((entry) => {
+							if (entry.isIntersecting && !played) {
+								void video.play();
```

Comment:
is it okay to call play` and `pause` every time the callback is called? maybe throw a `video.paused && void video.play()` and `!video.paused && void video.pause()` in there?

Reply:
Yeah fair point - added!